CONTRIBUTING.md

```diff
@@ -52,6 +52,35 @@ In addition to writing properly formatted commit messages, it's important to inc
 
 For package version upgrades and such a one-line commit message is usually sufficient.
 
+## Rebasing between branches (i.e. from master to staging)
+
+From time to time, changes between branches must be rebased, for example, if the
+number of new rebuilds they would cause is too large for the target branch. When
+rebasing, care must be taken to include only the intended changes, otherwise
+many CODEOWNERS will be inadvertently requested for review.  To achieve this,
+rebasing should not be performed directly on the target branch, but on the merge
+base between the current and target branch.
+
+In the following example, we see a rebase from `master` onto the merge base
+between `master` and `staging`, so that a change can eventually be retargeted to
+`staging`. The example uses `upstream` as the remote for `NixOS/nixpkgs.git`
+while the `origin` remote is used for the remote you are pushing to.
+
+
+```console
+# Find the common base between two branches
+common=$(git merge-base upstream/master upstream/staging)
+# Find the common base between your feature branch and master
+commits=$(git merge-base $(git branch --show-current) upstream/master)
+# Rebase all commits onto the common base
+git rebase --onto=$common $commits
+# Force push your changes
```

Comment:
```suggestion
# Force push your changes
git fetch upstream staging
git rebase upstream/staging
```

Reply:
this shouldn't be necessary since they should have done `git rebase --onto=$common $commits`

Reply:
Rebasing onto upstream/staging should only be done after having force-pushed *and* switched the target branch in GitHub (see https://github.com/NixOS/nixpkgs/pull/166472#discussion_r838982527). (I may be misunderstanding your suggestion though)

Reply:
The merge base between master and staging is usually older than staging itself, and in between that diff there may be conflicting changes still.

Reply:
oh yea, that's fair. Keep forgetting about divergence :(